proposals/interface-types/working-notes/choices/webidl.md

```diff
@@ -0,0 +1,257 @@
+# Interface Types and WebIDL
+
+## Preamble
+This note is part of a series that explain some of the major design choices in
+the Interface Types specification.
+
+## Introduction
+
+A key motivation for Interface Types is the semantic gap between the
+expressivitiy of core webAssembly and the needs of API designers and application
+and library authors. Part of this gap is represented by the difference between
+the type system of core webAssembly and the needs of a typical programmer.
+
+As an example of this, core webAssembly has no intrinsic type corresponding to
+`string`; even though this is arguably the most important type in many if not
+most applications.[^The reasons for this lack are beyond the scope of this
+note.] Note that neither the MVP of webAssembly nor the vision implied by the
+[GC proposal][GC] have `string` as an intrinsic type.
+
+A core part of this proposal is the design of an IDL (Interface Definition
+Language); one goal of which is to address the semantic gap. However, in
+designing an IDL there remain several questions, which this note aims to
+address:
+
+* What are the design considerations that lie behind the specification of the
+  Interface Types IDL?
+* Why design a new IDL, why not select one of the available IDLs?
+* Why not WebIDL?
+* Are we intending to support interoperability between different programming languages?
+* Given the importance of the web platform, what is the relationship between the Interface Types IDL and WebIDL?
+
+## Intended Scope of Interface Types
```

Comment:
This section feels out of place, in that it doesn't (in my reading) have bearing on the question of WebIDL. Rather I take it to be reiterating the motivation section of the Explainer.

My first recommendation would be to simply remove this entire section and its subsections. If that sounds wrong to you, it'd be useful to hear what you intended by including it here, so that I can help offer more specific suggestions (I'm not adding any other comments to this section for now).

Reply:
I think some background to why we are doing IT is important when understanding this question. It is not simply a reiteration; it picks up on one aspect of the explainer motivation.

Reply:
As I asked above, could you shed light on what, in particular, you think this adds? "some background" is too fuzzy for me to grok, since to my eyes (even when putting myself in the shoes of someone with no familiarity with IT), this section just gets in the way of the meat of the document.

Reply:
Maybe the title is confusing.
Essentially, I am setting up an argument that goes like this:

We don't want to adopt your solution:
This is our problem/requirement
This is why your solution does not work
This is how our solution can work with your solution

Reply:
It's not this title, I've also read through the section. I think the issue for me is that the sub-sections seem like they could just as easily apply to the non-JS parts of WebIDL. I.e., there's the same lack-of-trust between author code and web APIs, and function calls are also the mechanism that WebIDL makes available for accessing web APIs.

I think the "meat" of the "why your solution does not work" is that WebIDL is so tied to JS (and that is indeed the argument made here). But I don't think WebIDL is insufficient "ownership-wise" (even if AllowShared is maybe a little nudge in that direction), nor does "interop via function calls" rule out WebIDL.

To my reading, the intro up to this point, plus the longer explainer (which I'd expect folks to at least skim first) make pretty clear what IT's goals are, do a sufficient goal of fulfilling the need for "This is our problem/requirement".

Reply:
Agreed with Adam here; this does feel like a diversion from the target of this document.